src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class SettingsStorageHelper
+{
+public:
+    SettingsStorageHelper(const QString &keyName)
+        : m_keyName(keyName)
+    {
+    }
+
+    void set(const T &value)
+    {
+        SettingsStorage::instance()->storeValue(m_keyName, value);
+    }
+
+    T get(const QVariant &defaultValue = QVariant())
+    {
+        return SettingsStorage::instance()->loadValue(m_keyName, defaultValue).template value<T>();
+    }
+
+private:
+    QString m_keyName;
```

Comment:
>This is my naíve attempt to protect from the code like

But that is not likely to happen...

what about like this, is there going to be any problems?
```c++
explicit StorableValue(const std::string &keyName, const QVariant &defaultValue = QVariant())
        : m_keyName(keyName) {}
const std::string m_keyName;
```